### ISSUE TITLE
Submodule and sub-directory project is broken

### ISSUE THREAD

**Issue by bersace** [mgedmin/check-manifest#153]
Hi,

Say you have a repository with a Python project in a subdirectory and a submodule :

```
top_srcdir/
                /.gitmodules
                /project/
                            /setup.py
                            /MANIFEST.in
                            /submodule
```

check-manifest 0.47 project/ fails to determine the presence of a submodule becase `_has_submodules` checks for `project/.gitmodules` instead of top level .gitmodules.

A workaround is to put an empty `project/.gitmodules` file.

May I suggest to always call `ls-files` with `--recurse-submodules` ?

Thanks for check-manifest.

**Comment by mgedmin** [comment]
The only reason I'm not always passing `--recurse-submodules` to `git ls-files` is that it used to be *horrendously* slow on Windows back when I maintained my own Jenkins builds on a Windows VM.  Builds literally took *hours*.

I no longer maintain a Jenkins, and I believe newer Git versions have rewritten the submodule recursion code in C rather than the shell script that it used before, so should be fine to remove that workaround.

**Comment by mgedmin** [comment]
Released check-manifest 0.48 with a fix for this.

**Comment by bersace** [comment]
You're awesome. I had hard time too maintaining Jenkins CI. Kudo !
